src/Countries/Malaysia.php

```diff
@@ -0,0 +1,540 @@
+        ];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionSabah(int $year): array
+    {
+        return [
+            'Tahun Baru' => $this->newYear(),
+            'Pesta Kaamatan' => $this->pestaKaamatan($year),
+            'Pesta Kaamatan Hari Kedua' => $this->pestaKaamatan($year)->addDays(1),
+            'Good Friday' => $this->goodFriday($year),
+            'Harijadi Yang di-Pertua Negeri Sabah' => CarbonImmutable::parse("first saturday of october {$year}", $this->timezone),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+            'Hari Natal' => CarbonImmutable::createFromDate($year, 12, 24, $this->timezone),
+        ];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionSarawak(int $year): array
+    {
+        return [
+            'Tahun Baru' => $this->newYear(),
+            'Hari Sarawak' => CarbonImmutable::createFromDate($year, 7, 22, $this->timezone),
+            'Hari Gawai' => CarbonImmutable::createFromDate($year, 6, 1, $this->timezone),
+            'Hari Gawai Kedua' => CarbonImmutable::createFromDate($year, 6, 2, $this->timezone),
+            'Harijadi Yang di-Pertua Negeri Sarawak' => CarbonImmutable::parse("second saturday of october {$year}", $this->timezone),
+            'Good Friday' => $this->goodFriday($year),
+        ];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionSelangor(int $year): array
+    {
+        return [
+            'Tahun Baru' => $this->newYear(),
+            'Hari Thaipusam' => $this->hariThaipusam($year),
+            'Hari Nurul Al-Quran' => $this->nuzulQuran($year),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+            'Hari Keputeraan Sultan Selangor' => CarbonImmutable::createFromDate($year, 12, 11, $this->timezone),
+        ];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionTerengganu(int $year): array
+    {
+        return [
+            'Israk dan Mikraj' => $this->israkMikraj($year),
+            'Hari Ulang Tahun Pertabalan Sultan Terengganu' => CarbonImmutable::createFromDate($year, 3, 4, $this->timezone),
+            'Hari Nurul Al-Quran' => $this->nuzulQuran($year),
+            'Hari Keputeraan Sultan Terengganu' => CarbonImmutable::createFromDate($year, 4, 26, $this->timezone),
+            'Hari Arafah' => $this->hariArafah($year),
+            'Hari Raya Aidiladha Hari Kedua' => $this->hariAidiladha($year)->addDays(1),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+        ];
+    }
+
+    private function newYear(): string
+    {
+        return '01-01';
+    }
+
+    private function lunarNewYear(int $year): CarbonImmutable
+    {
+        return $this->chineseToGregorianDate('01-01', $year);
+    }
+
+    protected function getIslamicFormatter(): IntlDateFormatter
```

Comment:
If possible, I prefer this in a separate trait, just like the ChineseCalender.php